src/mds/MDCache.cc

```diff
@@ -3555,6 +3555,82 @@ void MDCache::remove_inode_recursive(CInode *in)
   remove_inode(in);
 }
 
+/**
+ * For all inodes, dirs, dentries below an inode, compose expiry
+ * messages.  This is used when giving up all replicas of entities
+ * for an MDS peer in the 'stopping' state, such that the peer can
+ * empty its cache and finish shutting down.
+ *
+ * We have to make sure we're only expiring un-referenced items to
+ * avoid interfering with ongoing stray-movement (we can't distinguish
+ * between the "moving my strays" and "waiting for my cache to empty"
+ * phases within 'stopping')
+ *
+ * @return false if we completed cleanly, true if caller should stop
+ *         expiring because we hit something with refs.
+ */
```

Comment:
As discussed in standup, this is part of our inherited Google style guide
and I think all our new code that's well-behaved is doing this. :)